src/qibo/models.py

```diff
@@ -169,7 +193,7 @@ def __init__(self, circuit, hamiltonian):
         self.circuit = circuit
         self.hamiltonian = hamiltonian
 
-    def minimize(self, initial_state, method='Powell', options=None, compile=True):
+    def minimize(self, initial_state, method='Powell', options=None, parallel=None, compile=True):
```

Comment:
If I understood this correctly, here `parallel` has to be a dictionary that contains the `"max_workers"` key and maps it to an integer. Wouldn't be simpler in terms of usage if `parallel` was just an integer and perhaps was renamed to `max_workers` to be more explicit? Unless this dictionary also controls something else that I am missing.

Either way, I guess that once we add documentation it will be clear what `parallel` is from the user perspective.